### ISSUE TITLE
Create a landing page for new Matrix users

### ISSUE THREAD

**Issue by noahpierau** [decred/dcrweb#452]
Matrix is an open source communication platform that has gained popularity among Decred contractors. However, it is poorly understood by most new users. The idea is to create a landing page similar to slack.decred.org to educate people about this platform.

There are different ways to register on Matrix. The landing page should educate people about the pros and cons of registering on the Decred homeserver instead of the default Matrix homeserver. Some advantages: 1) you have a nice decred.org in your user ID. 2) if the Decred homeserver gets disconnected from other homeservers, its users will still be able to communicate with one another. 3) easier to find other Decred rooms and users. 4) you won't route through a machine controlled by a 3rd party. There is one big disadvantage: when the user gets logged out and tries to reconnect, they will wonder why their password is invalid (they need to set their homeserver to the correct one with each login). This section should include a registration link with the custom homeserver filled out: https://riot.im/app/#/register?hs_url=https://matrix.decred.org

There are also different ways of reaching the Decred rooms. We should link to a collection of different Matrix clients, so users can pick their favorite: https://matrix.org/docs/projects/try-matrix-now.html and https://matrix.org/docs/projects/clients-matrix

If users want to use an existing Matrix account, the landing page should point them to https://matrix.to/#/+decred:decred.org

**Comment by degeri** [comment]
Some links that can be used by user to login. 

https://riot.im/app/#/login?hs_url=https://matrix.decred.org

https://matrix.to/#/+decred:decred.org?hs_url=https://matrix.decred.org (don't know if this is the correct way to use matrix links but it seems to work)

**Comment by xaur** [comment]
Something to consider about end-to-end encryption used in Matrix: it does not encrypt quite a lot of metadata: who talks to whom, when, in what room, maybe something else.

**Comment by noahpierau** [comment]
A while ago I created this image as a guide to sign up for Matrix:

![image](https://user-images.githubusercontent.com/16287891/49689189-f225ff80-fb1d-11e8-9412-a7c64c297879.png)

It helped people to sign up "correctly" so maybe a similar guide can be included on the new page.

**Comment by s-ben** [comment]
@ZeroASIC and I were talking about collaborating on this. I would create some copy and he would work on a page for decred.org. @noahpierau , this image is really good. I'm wondering how we can reuse it. My concern now, upon reflection, is that a) detailed instructions may be too long to include on decred.org?, and that b) I know there's been a general redesign afoot, and Ditto is coming up with new messaging as we speak. I'm worried we will produce something that doesn't fit well into decred.org. Thoughts? I do think that it's worth creating a separate page for Matrix. Just wondering what form it should take. E.g. should we move the detailed signup instructions with screenshots to dcrdocs, and make the decred.org landing page more of a summary?

**Comment by s-ben** [comment]
@ZeroASIC , have taken a stab at copy for the landing page. I'm thinking we could do something similar to [https://slack.decred.org/](https://slack.decred.org/), but without the email registration field?

## Join our Matrix community

Matrix is a secure, open messaging platform popular with Decred contractors. It allows users to communicate with other community members across other Decred channels, such as Slack, Telegram, and Discord, without leaving the Matrix chat client. Matrix requires only an email or phone number for registration. 

To join Matrix, you’ll need to choose a client and a homeserver. Matrix supports numerous clients and homeservers, but for first-time users, we recommend the most well-supported Matrix client, Riot, and Decred’s homeserver. Using the Decred homeserver makes it easier to find Decred chat rooms, puts ‘decred.org’ in your ID, and allows you to continue messaging should Decred’s homeserver become disconnected with other homeservers.

To register using the Riot web app, with Decred’s homeserver information pre-filled, click the link below. 
https://riot.im/app/#/register?hs_url=https://matrix.decred.org

_Note: you must register with an email address if you want password resets._ 

If you already have an existing Matrix account, use the link below to log into Decred’s homeserver. 
https://matrix.to/#/+decred:decred.org
 
To browse other available Matrix clients and home servers, see the Matrix Getting Started page,
https://matrix.org/docs/projects/try-matrix-now.html

**Comment by ZeroASIC** [comment]
I started a draft for this in a new branch (https://github.com/ZeroASIC/dcrweb/tree/issue452)

I added two directories for the community landing pages, one for Matrix and an update for Slack because the menu was out of date.  They're not link directly from the community page yet.  The Matrix page is based on the existing layout for slack.decred.org so they share a background image.

**Comment by noahpierau** [comment]
Can you share a screenshot of the WIP?

**Comment by ZeroASIC** [comment]
![screenshot_2018-12-11_16-40-29](https://user-images.githubusercontent.com/4195716/49835212-18d57780-fd64-11e8-8500-a21f2e4077c7.png)

**Comment by noahpierau** [comment]
Awesome, this is exactly what we need! 

Really great job everyone, I think we should strive to publish this page ASAP.

**Comment by ZeroASIC** [comment]
If no one has any changes for it, I can update the Community page and do a Pull Request.  Should I leave the /slack/index.html where it is or just delete it?  All I did was update the menu and some paths.

**Comment by s-ben** [comment]
Looks good! 

I would just make a couple suggestions. 

- To be more consistent with the [slack page](https://slack.decred.org/), I would remove the `Matrix is a secure, open messaging system. Hope to see you soon!` from the left pane. Keep the left pane as just signup instructions like the slack page. We also are repeating the same text on the first line of the description in the right pane.
- For the _Note: you must register with an email..._, the italics in my original draft look kinda off now. To be consistent with the slack page, I suggest we use **bold** for emphasis.
- For the second link, we should have link text like the other links. I'm thinking `Login to Decred homeserver`

**Comment by ZeroASIC** [comment]
![screenshot_2018-12-11_22-29-15](https://user-images.githubusercontent.com/4195716/49847186-7681b800-fd94-11e8-9fc5-a23081ad02b6.png)

**Comment by ZeroASIC** [comment]
Should I update the footer to match the rest of the site?  I could either update both of them or leave this one as is.

**Comment by s-ben** [comment]
Maybe more of a call for @noahpierau or other design peeps, but I kinda like it as is. I searched a few other pages and it doesn't appear we're entirely consistent with the footer. Though whether this is by design or accident I'm not sure.

**Comment by ZeroASIC** [comment]
That's an accident.  I thought I fixed that once, but I'm not the only one doing things.  /press/ is a bit off.  Other domains are completely different.

**Comment by s-ben** [comment]
It's like dcrdocs. An amorphous decentralized cloud of content morphing into something coherent.

**Comment by jrick** [comment]
There's still a giant slack logo in the background picture.

**Comment by xaur** [comment]
I worry that people will skip the text due to distracting background image. Personally I'm more likely to read simple black on white text.

**Comment by xaur** [comment]
> Matrix is a secure

Too abstract, what do you mean by secure?

> It allows users to communicate with other community members across other Decred channels, such as Slack, Telegram, and Discord, without leaving the Matrix chat client.

Bridge software allows it, not Matrix.

> Matrix requires only an email or phone number for registration.

Email and phone are optional and this should be emphasized. This is a very cool and very rare feature on the web. You can choose to have a _first party registration_ that doesn't involve any middlemen (email service). But users must be notified of pros and cons of setting or not setting an email.

**Comment by ZeroASIC** [comment]
I'd like to keep this page as similar to slack.decred.com as possible.  Anything we change about the design should get propagated back to that page, which is why I made a copy of it in /slack/.  I think what we really need is a short term solution and a long term plan.  In the short term it doesn't need to be perfect, but it does need to be effective.  Eventually we could make landing pages for all of the supported chat platforms.

**Comment by ZeroASIC** [comment]
> Too abstract, what do you mean by secure?
There's some danger in providing too much information.  I'm not usually in favor of bombarding website users with lots of technical information.  That information usually belongs in a technical document, not the copy intended to bring users into the community.  We could make an additional document to accompany this page in docs.decred.org (e.g. Matrix faq) and provide a link to it.

**Comment by s-ben** [comment]
Agree with @ZeroASIC on need for keeping web copy concise. @xaur , I was imagining your criticisms while I was writing copy, lol.  I think the best solution would be to: 1) edit copy to incorporate xaur points (while keeping it high-level), 2) link to a separate page that provides more details on security, accompanied by a line like, "to read more about security details of Matrix, click here".

I'm tempted to create a page on dcrdocs on Matrix. However, I don't see a non-awkward place to put it. Creating a new page might require some restructuring of other pages/menu. That might be worth doing anyway, but it's something to consider. Is there an outside page we can link to that summarizes this info for users?

**Comment by s-ben** [comment]
Also, is there another image we can use that doesn't have the Slack logo? At the very least, if there's no other image handy (or Design can't be bothered), we could Photoshop out the '#' logo, maybe replace it with Matrix logo. Wouldn't take much time to do.

**Comment by noahpierau** [comment]
The way I look at it: the "logo" is really just a hash tag, which is used in Matrix as well.

Maybe we can link to the Matrix docs?

Agreed with updating the copy, and maybe editing in the Matrix logo, but IMO we shouldn't spend too much time on it. The most important thing is that we put it live. After that we can start iterating on it.

**Comment by ZeroASIC** [comment]
> No need to bombard the user with any techincal details, I suggest to just remove the claim about security unless you can link to a page that explains why it is "secure".

This is the official doc on encryption.  We don't have to use it, I just wanted to point it out.
https://matrix.org/docs/guides/faq#encryption

**Comment by s-ben** [comment]
Have updated the copy on the first paragraph to address concerns (below). I think it's OK to use "secure" in the first sentence as long as we provide a link to further details about security features.

"Matrix is a secure, open messaging platform popular with Decred contractors. It uses bridge technology to allow users to communicate with community members across other Decred channels such as Slack, Telegram and Discord, without leaving Matrix. Matrix offers first party registration; no identifying information or third party service is required to sign up. Users should be aware however that some features, such as password reset, do require an email address.  For more on Matrix’s security features and available clients, see the Matrix project [homepage](https://matrix.org/blog/home/)."

@ZeroASIC, just saw your link to encryption. This link is good, though perhaps the link to the homepage is better, as a security-minded user will be motivated to find this link themselves, and other users may just want general info (like what other clients are available).

**Comment by ZeroASIC** [comment]
I removed the background image so we could compare it to the first draft.
![screenshot_2018-12-13_14-46-30](https://user-images.githubusercontent.com/4195716/49966663-41857a80-fee6-11e8-983c-a043d97fcc72.png)

**Comment by s-ben** [comment]
Looks good!

![image](https://user-images.githubusercontent.com/1634777/49967574-257adc80-fed8-11e8-9a74-181b0136f205.png)

Note: just now, GitHub just started giving me a vague error message when I tried to upload an image (something went wrong, try again?). After trying a few tries, I turned off Brave's tracking (Shields Down). Image uploaded fine... 

![image](https://user-images.githubusercontent.com/1634777/49968116-bb633700-fed9-11e8-9985-654ccac70b67.png)

**Comment by xaur** [comment]
If it's not too late I'd simplify 2nd sentence

> With the help of bridge software Matrix users can communicate with users on other Decred platforms such as Slack and Discord.

Also dropped Telegram because it is no longer bridged.

**Comment by ZeroASIC** [comment]
I updated the second sentence, replaced the Roit.im link with Matrix on the community page.  Is Rocket.Chat bridged?  If not then I should move it on the community page.

**Comment by xaur** [comment]
@ZeroASIC there are talks to drop RocketChat because almost nobody uses it. I suggest to not mention it until we know for sure if it will stay or not.

**Comment by ZeroASIC** [comment]
@xaur That's fine, I wasn't going to change the copy.  I'm just reordering the links on the community page.  I think I'll put Rocket.Chat on the second row since it might get dropped.

**Comment by ZeroASIC** [comment]
I'm going to put the pull request in this afternoon.  I think we've had enough iterations in development that I'm starting to get into the details more than I need to for an initial release.  I can always update it again later.

**Comment by ZeroASIC** [comment]
Pull Request #462 is in.

**Comment by s-ben** [comment]
Having just the circular directory icon looks better than the screenshot I suppose, but it's still slightly awkward visually, and redundant if we're using the more descriptive text IMO. Maybe if we can make the circle fit better with text (smaller maybe?), we should keep it. But I would vote to just delete it and go with just text.

**Comment by ZeroASIC** [comment]
Works for me, just text then.  Would a link to the directory listing be useful?

**Comment by s-ben** [comment]
Good idea, but I'm wondering if it will be a different link based on whether they choose the Decred custom homeserver or not, or if they use a different client. Could be multiple points of entry.

**Comment by ZeroASIC** [comment]
Yeah, the complication with this section is that we're favoring one configuration on one piece of software.

**Comment by xaur** [comment]
Link to directory listing: https://riot.im/app/#/directory but it indeed may show different results depending on homeserver.

A more stable link is Decred matrix community: https://riot.im/app/#/group/+decred:decred.org - it's missing couple rooms but that will be fixed eventually.

Re the round button, you can make a rectangular screenshot of the button panel with the two buttons highlighted.

**Comment by s-ben** [comment]
Just tested out the registration flow with a new phone number (FYI, Google Voice gives you a free number, though you have to verify it with a text to an existing number), used our registration link that auto-fills the Decred homeserver and suggested directory link above (https://riot.im/app/#/group/+decred:decred.org). It all worked good. Honestly, I think we're fine directing people towards Riot+homeserver+directory link. Matrix is _**not**_ user friendly (yet). I definitely struggled with it. We should offer the easiest, safe path on. If someone is hardcore enough to want to choose different Matrix servers, etc., they're technically savvy enough not to need these high-level instructions. Let's ship this puppy!
